pkg/webui/components/dropdown/portal/index.js

```diff
@@ -0,0 +1,66 @@
+// Copyright © 2024 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React, { useEffect } from 'react'
+import DOM from 'react-dom'
+
+import PropTypes from '@ttn-lw/lib/prop-types'
+
+// This component create a portal to render children into a different part of the DOM.
+const Portal = ({ children, visible, positionReference }) => {
+  const [buttonPosition, setButtonPosition] = React.useState()
+  useEffect(() => {
+    if (positionReference.current && visible) {
+      const rect = positionReference.current.getBoundingClientRect()
+      setButtonPosition(rect)
+    }
+  }, [positionReference, visible])
+
```

Comment:
```suggestion
if (!buttonPosition) {
  return null
  }
```

Otherwise the dropdown can briefly flash at the top left.